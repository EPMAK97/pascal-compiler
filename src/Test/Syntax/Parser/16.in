function sum(a, b : integer; var c : array of integer) : double;
Begin
    result := a + b;
end;
Begin
end.