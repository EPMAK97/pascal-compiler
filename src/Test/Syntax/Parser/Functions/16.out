FUNCTION : test
    result : NIL
    function vars {
    real : type DOUBLE
    rec : type RECORD
        a : DOUBLE
    END
    FUNCTION : f
        result : RECORD
        function params {
        a : DOUBLE
        b : DOUBLE
        c : DOUBLE
        d : RECORD
                a : DOUBLE
        END
        }
        function vars {
        }
        └── statements
                     ├── :=
                     |    ├── .
                     |    |   ├── d
                     |    |   └── a
                     |    └── /
                     |        ├── +
                     |        |   ├── a
                     |        |   └── b
                     |        └── c
                     └── :=
                          ├── Result
                          └── d

    r : RECORD
        a : DOUBLE
    END
    }
    └── statements
                 └── :=
                      ├── r
                      └── f
                          └── params
                                   ├── 1.0
                                   ├── 10.1
                                   ├── 3
                                   └── r

